scribus/plugins/import/pdf/slaoutput.h

```diff
@@ -153,6 +153,73 @@ class AnoOutputDev : public OutputDev
 	QStringList *m_importedColors;
 };
 
+/* PDF TextBox Framework */
+/*
+* Holds all the dtails for each glyph in the text imported from the pdf file.
+*
+*/
+struct PdfGlyph {
+	QPointF position;    // Absolute glyph coords
+	double dx;  // X advance value
+	double dy;  // Y advance value
+	double rise;    // Text rise parameter
+	QString code;   // UTF-16 coded character but we only store and use UTF-8, the slternstive is const char * for utf8 so far as qt is concerned
+	bool is_space;
+};
+
+
+class TextRegionLine
+{
+public:
+	qreal maxHeight = -1;
+	qreal modeHeigth = -1;
+	qreal width = -1;
+	int glyphIndex = -1;
+	QPointF baseOrigin = QPointF(-1, -1);
+	std::vector<TextRegionLine> segments = std::vector<TextRegionLine>();
+
+};
+
+class TextRegion {
+public:
+	enum FRAMEWORKLINETESTS {
+		FIRSTPOINT,
+		SAMELINE,
+		STYLESUPERSCRIPT,
+		STYLENORMALRETURN,
+		STYLEBELOWBASELINE,
+		NEWLINE,
+		ENDOFLINE, //TODO: Implement an end of line test
+		FAIL
+	};
+
+	QPointF textRegioBasenOrigin = QPointF(-1, -1);
+	qreal maxHeight = -1;
+	qreal modeHeigth = -1;
+	qreal lineSpacing = -1;
+	std::vector<TextRegionLine> textRegionLines = std::vector<TextRegionLine>();
+	qreal maxWidth = -1;
+	QPointF lineBaseXY = QPointF(-1, -1); //updated with the best match left value from all the textRegionLines and the best bottom value from the textRegionLines.segments;
+	QPointF lastXY = QPointF(-1, -1);
+	static bool coLinera(qreal a, qreal b);
+	bool closeToX(qreal x1, qreal x2);
+	bool closeToY(qreal y1, qreal y2);
+	bool adjunctLesser(qreal testY, qreal lastY, qreal baseY);
+	bool adjunctGreater(qreal testY, qreal lastY, qreal baseY);
+	TextRegion::FRAMEWORKLINETESTS linearTest(QPointF point, bool xInLimits, bool yInLimits);
+	TextRegion::FRAMEWORKLINETESTS isRegionConcurrent(QPointF newPoint);
+	TextRegion::FRAMEWORKLINETESTS moveToPoint(QPointF newPoint);
+	TextRegion::FRAMEWORKLINETESTS addGlyphAtPoint(QPointF newGlyphPoint, PdfGlyph new_glyph);
+	void renderToTextFrame(PageItem* textNode, ParagraphStyle& pStyle);	
+	std::vector<PdfGlyph> glyphs; //this may replace some of the other settings or it may not, certainly not font as text gets flushed if the font changes

```

Comment:
always prefix this kind of comments with your name (or abbreviation) and maybe add a date (even if blame can tell us)

Reply:
comment removed